### ISSUE TITLE
Get-ChildItem -Recurse -Name /path/glob.* does not recurse

### ISSUE THREAD

**Issue by rkitover** [PowerShell/PowerShell#16477]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

```powershell
gci -r -n /windows/*.dll
```

### Expected behavior

I would expect that do the same thing as

```powershell
gci -r /windows/*.dll
```
,
but only give me the file names.

### Actual behavior

Only the files at the location are listed, and not any locations below that path.

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.2.0
PSEdition                      Core
GitCommitId                    7.2.0
OS                             Microsoft Windows 10.0.22499
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by jborean93** [comment]
The way this is working is that you are calling `Get-ChildItem -Path /windows/*.dll -Recurse` and it will first enumerate the "children" of files matching your wildcard, i.e. `*.dll` in that directory. Each of these objects will be files with nothing to recurse into. What you want to do is

```powershell
Get-ChildItem -Path /windows -Recurse -Include *.dll -Name
````

The `-Include` option is also part of an example to get some more information https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/get-childitem?view=powershell-7.2#example-4--get-child-items-using-the-include-parameter. Technically `-Filter` will be more efficient for Windows file systems

**Comment by rkitover** [comment]
Thank you for the explanation, so I can shorten this to:

```powershell
gci -r -n /windows -i *.dll
```

which is not terrible.

This also works consistently without `-n`, which is good.

This serves the purposes of the document I'm working on, however, I still believe that the behavior I described is not consistent from the perspective of the user.

By the way, and not related to this issue, my document is here:

https://github.com/rkitover/windows-dev-guide

and I would appreciate any feedback from you guys.

**Comment by MartinGC94** [comment]
> Technically `-Filter` will be more efficient for Windows file systems but it doesn't work well with `-Recurse` so I wouldn't recommend it here.

I've never had any issues with this. Do you have any examples where it wouldn't work?

**Comment by jborean93** [comment]
I was testing on Linux and it didn't work as expected but on Windows it does which makes sense because `-Filter` is done at the Windows file system driver level and may not necessarily be the same on Linux.

**Comment by iSazonov** [comment]
> Technically -Filter will be more efficient for Windows file systems

Today no but we could improve this if FileSystem provider improvements  will be approved for next milestone.

**Comment by iSazonov** [comment]
Dup #9014
